docs/blueprints/EP031.rst

```diff
@@ -0,0 +1,273 @@
+
+To illustrate the challenge, consider an EVC terminating on INT Hop Z on port 23. The user packet with INT metadata comes from port 11. **mef_eline** would create the following flows (for simplicity, just one direction is presented):
+
+  0. **met_eline** default behavior:
+
+    - match:
+
+      - priority: DEFAULT (32678)
+      - match: <in_port=11,vlan_vid=20>
+
+    - instruction:
+
+      - action [set_queue, pop_vlan, etc.]
+      - action: output to port 23.
+
+To enable INT, first a physical loop has to be deployed. For this example, on INT Hop Z, port 1 is connected to port 2 by a physical fiber patch cord (done during commissioning). Then, the following flows need to be ADDED to the pipeline:
+
+  1. Adding INT metadata:
+
+    - match:
+
+      - table 0
+
+        - priority: Higher (40000)
+        - match: <in_port= **11**,vlan_vid=20,eth_type=0x800,ip_proto=6>  # TCP
+        - instrutions:
+
+          - action: add_int_metadata
+          - action [set_queue, pop_vlan, etc.]
+          - action: output port **1** (loop)
+
+        - priority: Higher (40000)
+        - match: <in_port= **11**,vlan_vid=20,eth_type=0x800,ip_proto=17>. # UDP
+        - instrutions:
+
+          - action: add_int_metadata
+          - action [set_queue, pop_vlan, etc.]
+          - action: output port **1** (loop)
+
+  2. Send Report and pop INT data (traffic is coming from port 2 that's the loop with port 1). Only INT data gets into the loop.
+
+    - match:
+
+      - table 0
+
+        - priority: Higher (40000)
+        - match: <in_port= **2**,vlan_vid=20>
+        - instrutions:
+
+          - action: send_report
+          - action go to table 2
+
+
+      - table 2
+
+        - priority: Higher (40000)
+        - match: <in_port= **2**,vlan_vid=20>
+        - instrutions:
+
+          - action: pop_int
+          - action [set_queue, pop_vlan, etc.]
+          - action: output port **23** (original port)
+
+
+IV. How to enable INT for EVCs
+==============================
+
+The goal for the **telemetry** app is to enable telemetry for ALL EVCs. However, it must support enabling and disabling telemetry for a single EVC or ALL EVCs. This is the approach:
+
+  1 . The **telemetry** napp will start operating once **mef_eline** is loaded and EVCs and their flows are pushed to the data plane.
+  2. **telemetry** will listen for events *kytos/mef_eline.(redeployed_link_(up|down)|deployed)* and *kytos.mef_eline.created* issued by **mef_eline**.
+  3. For each EVC identified, **telemetry** will
+    1. use EVC's cookie to get all flow entries created by **flow_manager** IF telemetry is not already enabled.
+    2. push more specific flows as described in Section II
```

Comment:
I was considering using the same cookie. Since the first 2 bits (or bytes) of the cookie are reserved, can I use one bit/byte to assign ownership to the telemetry napp? (I am not calling it INT napp but we can change it).

Reply:
Yes. If we could use `0xab` (that currently of_lldp) uses that'd make it easier to mask with `0xaa`, and then allocate a new one for `of_lldp` (and provide an script to update existing installed entries), let's see if `table_id: OFPTT_ALL` deletes efficiently. 

 In the future, when we also have P4 NApps we could potentially keep the same semantics on a `TableEntry` with [`metadata: bytes`](https://github.com/p4lang/p4runtime/blob/v1.3.0/proto/p4/v1/p4runtime.proto#L204) storing the existing cookie byte there.

Reply:
@viniarck , thanks for the comments and suggestions. Please, define the approach for the telemetry napp to use. I am ok with keeping the same cookie, using an adjacent value, or a prefix.

Reply:
I reserved Section X for documenting the cookie strategy.

Reply:
@jab1982, let's go with a new value and not adjacent one. Although we could have benefit from deleting flows with a shared prefix, as long as batched/paced deletions are implemented it should be all set, and arguably easier to debug, equivalent to what was brought up on this thread https://github.com/kytos-ng/kytos/pull/279/files#r958512455.